Includes/Rosetta/Managers/TriggerEventHandler.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2019 Chris Ohk, Youngjoong Kim, SeungHyun Jeon
+
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#ifndef ROSETTASTONE_TRIGGER_EVENT_HANDLER_HPP
+#define ROSETTASTONE_TRIGGER_EVENT_HANDLER_HPP
+
+#include <functional>
+
+namespace RosettaStone
+{
+class Entity;
+
+//!
+//! \brief TriggerEventHandler class.
+//!
+//! This is an event handler class for trigger.
+//!
+class TriggerEventHandler
+{
+ public:
+    using Func = std::function<void(Entity*)>;
+
+    //! Default constructor.
+    TriggerEventHandler();
+
+    //! Constructs trigger event handler with given \p func.
+    //! \param func A functor to run.
+    TriggerEventHandler(Func func);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'TriggerEventHandler' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=5011091)